static/js/user_status_ui.js

```diff
@@ -47,6 +48,7 @@ exports.submit_new_status = function () {
 
     user_status.server_update({
         status_text: new_status_text,
+        away: old_status,
```

Comment:
I don't think the solution here is to send the `old_status` as part of the server update, as that can lead to us accidentally sending stale data.  I think the problem here is more on the server side.

Reply:
Okay, I will look into this again if I can find the bug on the server side.